scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/DefaultPacBio.java

```diff
@@ -353,4 +353,9 @@ private void processMetadata(Document metadata, PacBioNotificationDto dto) {
   private static boolean isStringBlankOrNull(String s) {
     return s == null || "".equals(s.trim());
   }
+
+  @Override
+  public boolean isFilePathValid(File fs_obj) {
+    return fs_obj.isDirectory() && fs_obj.canExecute() && fs_obj.canRead();
```

Comment:
Can put this in the abstract class, Javadoc the default behaviour, and only override if it should do something different (DefaultPromethion)

Reply:
When the Default MinION Processor is implemented, we will have a 50/50 split between reading directories vs files. Should directories still be considered default?

Reply:
Maybe it would be better to add a `PathType` enum `{DIRECTORY, FILE}` and `getPathType` abstract method in `RunProcessor`. Then the `isFilePathValid` method can be only in the abstract class, and check `getPathType` to determine which checks to do. This reduces duplication so that if, for example, you realize another permission that needs checked on run directories, you only need to make that change in one place.